tool/src/main.rs

```diff
@@ -54,12 +58,25 @@ fn main() {
         tas = Tas::new().unwrap();
         println!("TAS created successfully.");
     }
-
-    let script_file = env::args().skip(1).next().unwrap_or("tas.lua".to_string());
-    println!("Executing Script {} ...", script_file);
-    tas.execute(script_file, &config);
+    let script_file = if env::args().len() == 1 {
+        if !std::env::current_dir().unwrap().join("tas.lua").is_file() {
+            let cur_dir = std::env::current_dir().unwrap();
+            let cur_dir = cur_dir.to_str().unwrap();
+            let result = nfd::open_file_dialog(Some("lua"), Some(cur_dir)).unwrap();
+            match result {
+                Response::Okay(file_path) => Path::new(&file_path).to_path_buf(),
```

Comment:
`file_path` is a String. You should create the `PathBuf` directly from that string, which removes one allocation: `PathBuf::from(file_path)`.